### PR TITLE
Homework05

### PR DESCRIPTION
**Pull request by AJIEIIIA**

### CODE REVIEW

File: course5/homework03/homework03.cc [hunk 1]
```diff
@@ -1,19 +0,0 @@
-#include <cstdio>
```
Comment (maciekgajewski) [inline course5/homework03/homework03.cc]: Why does the pull request for homework 5 contain changes to homework 3?

File: course5/homework05/.gitignore [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# This file is used to ignore files which are generated
```
Comment (maciekgajewski) [inline course5/homework05/.gitignore:1]: This is noise; this makes the change more difficult to review.

File: course5/homework05/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+project(homework05)
+cmake_minimum_required(VERSION 2.8)
```
Comment (maciekgajewski) [inline course5/homework05/CMakeLists.txt]: Our minimum is 3.0

File: course5/homework05/vector2d.h [hunk 4]
```diff
@@ -0,0 +1,51 @@
+#ifndef VECTOR2D
```
Comment (maciekgajewski) [inline course5/homework05/vector2d.h]: Use ```#pragma once``` instead.

File: course5/homework05/vector2d.h [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+#define VECTOR2D
+#include <ostream>
+#include <cstring>
+
+namespace homework05
+{
+using namespace std::literals;
```
Comment (maciekgajewski) [inline course5/homework05/vector2d.h]: You are not using any of the literals provided by the standard library, so this is not needed.

File: course5/homework05/vector2d.h [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    Vector2D(double x, double y);
+    Vector2D() = delete;
+    double GetX() const;
+    double GetY() const;
+    void SetX(double x);
+    void SetY(double y);
+    double GetR() const;
+    double GetPhi() const;
+
+    Vector2D operator + (const Vector2D& v);
```
Comment (maciekgajewski) [inline course5/homework05/vector2d.h]: Q: Can this operator modify the LHS object?

File: course5/homework05/homework05.cc [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+#include <cassert>
+#include <sstream>
+#include <cmath>
+#include "vector2d.h"
+
+using namespace homework05;
+
+void test_equals()
+{
+    auto v1 = Vector2D(1, 0);
```
Comment (maciekgajewski) [inline course5/homework05/homework05.cc]: Try ``const auto```

**Review by maciekgajewski — CHANGES_REQUESTED** [review]
Mostly good, but tests are only run on a non-const objects, which hides the fact that some of the methods that should be const, aren't.
Try to keep the signal/noise ratio higher.

File: course5/homework03/homework03.cc (hunk 1 above)
Comment (AJIEIIIA) [reply course5/homework03/homework03.cc]: removed

File: course5/homework05/vector2d.h (hunk 4 above)
Comment (AJIEIIIA) [reply course5/homework05/vector2d.h]: fixed

File: course5/homework05/vector2d.h (hunk 6 above)
Comment (AJIEIIIA) [reply course5/homework05/vector2d.h]: indeed. added const

File: course5/homework05/homework05.cc (hunk 7 above)
Comment (AJIEIIIA) [reply course5/homework05/homework05.cc]: added const to methods that do not modify object

File: course5/homework05/vector2d.h (hunk 5 above)
Comment (AJIEIIIA) [reply course5/homework05/vector2d.h]: removed

File: course5/homework05/CMakeLists.txt (hunk 3 above)
Comment (AJIEIIIA) [reply course5/homework05/CMakeLists.txt]: fixed

**Pull request closed without merging**